server/src/controllers/vaccinationController.ts

```diff
@@ -0,0 +1,65 @@
+import { Request, Response } from "express";
+import { vaccinationService } from "../services";
+
+export const vaccinationController = {
+  async index(req: Request, res: Response) {
+    const { animal_id } = req.params;
+    const vaccines = await vaccinationService.getAllVaccinations(animal_id);
+    res.json(vaccines);
+  },
+
+  async show(req: Request, res: Response) {
```

Comment:
small suggestio, amybe renaming it to getAllAnimalVaccinations would make it even clearer. Not critical, just a naming improvement to consider.